src/python/pants/engine/native.py

```diff
@@ -610,6 +605,7 @@ def init_externs():
       context = ExternContext(self.ffi, self.lib)
       self.lib.externs_set(context._handle,
                            self.ffi_lib.extern_log,
+                           logger.getEffectiveLevel(),
```

Comment:
Should have @kwlzn take a look at this, but: I'm not sure that when the externs are created we actually know what the log level will be. Would need to confirm that the bootstrap options are actually being parsed before the externs are initialized.

Reply:
agree that relying on global state here might be errorprone - a quick integration test might be the best way to confirm/protect this?

Reply:
Multiple levels of global state, unfortunately... iirc parsing bootstrap options mutates python's log level?

Reply:
Added test